p2p/client.go

```diff
@@ -134,6 +144,18 @@ func (c *Client) startWithHost(ctx context.Context, h host.Host) error {
 		return err
 	}
 
+	c.logger.Debug("blocking blacklisted peers")
```

Comment:
Shouldn't gater be used (blacklist applied) before starting gossip/DHT/peer discovery?

Reply:
whoops